### PR TITLE
Benadryl uses 12 march24

### PR DESCRIPTION
**Pull request by goutamnaik007**
**Please review the code**
No. of new TC's added : 40
Module Name : Benadryl Uses
Test Report for all suites : https://drive.google.com/file/d/1f5Qn9z-1Pfv9a2Q4o0PbqatY4RYnktrG/view?usp=drive_link
Test report for new TC's : https://drive.google.com/file/d/1dOCwX3jyHG15uiechVGDnFpKppPFX8lh/view?usp=drive_link

### CODE REVIEW

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java [hunk 1]
```diff
@@ -171,4 +159,16 @@ public void checkLinks(String key, String expectedUrl) {
 			Assert.fail("Link not displayed");
 		}
 	}
+	
+	/**
+	 * Function to close signup lightbox
+	 */
+	public void closeSignUpPopup() {
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java:166]: @goutamnaik007 this function you added at the bottom from line 81 ?

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+
+import com.jnj.actions.Action;
+import com.jnj.base.BaseClass;
+
+public class BenadrylUsesPage extends BaseClass {
+	Actions actions;
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java]: @goutamnaik007 why this is here.??? are we using this anywhere???

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	 * Function to verify sysmptoms causes
+	 * 
+	 * @param symptom     pass symptom
+	 * @param cause       pass cause
+	 * @param description pass expected desc
+	 */
+	public void verifySymptomCasuses(String symptom, String cause, String description) {
+		String actualDesc;
+		WebElement ele = null;
+
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java]: @goutamnaik007 pls remove empty line

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+		boolean eleDisplayed = getSpanHeadings(heading).isDisplayed();
+		if (eleDisplayed) {
+			extentPassLog("Related Content header displayed : ", true);
+			String actualHeading = getSpanHeadings(heading).getText();
+			Action.printAndAssert(actualHeading, heading);
+			Action.explicitWait(articleName(article), 30);
+			String actualArticle = articleName(article).getText();
+			Action.printAndAssert(actualArticle, article);
+			Action.explicitWait(readMoreLink(readMore), 30);
+			Action.performActionwithExtentInfoLog(readMoreLink(readMore), "click",
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java:256]: @goutamnaik007 pls remove the empty line

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+	 */
+	public void verifyReferences(String heading, String refText, String link, String expectedUrl) {
+		String actualHeading = getHeadings(heading).getText();
+		Action.printAndAssert(actualHeading, heading);
+		boolean eleDisplayed = refText(refText).isDisplayed();
+		if (eleDisplayed) {
+			extentPassLog("Reference point displayed : ", true);
+			String actualRef = refText(refText).getText();
+			extentPassLog("Reference point : ", actualRef);
+			Action.performActionwithExtentInfoLog(refLinks(link), "click", "Clicking on : " + refLinks(link).getText());
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java:279]: @goutamnaik007 pls remove the empty line

File: BenadrylAutomation/src/test/java/com/jnj/testcases/BenadrylUsesTest.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+			home.closePrivacyPopup();
+			header.navigateSubMenuLink(menu, subMenu);
+			signup.closeSignUpPopup();
+			uses.verifyAllergiesDesc(section, heading, description);
+		} else {
+			throw new SkipException("Test skipped : " + testcase);
+		}
+	}
+	
+	@Test(priority = 3, dataProvider = "symptoms", dataProviderClass = BenadrylUsesProvider.class)
```
Comment (vaibhavNagvekarSJI) [inline BenadrylAutomation/src/test/java/com/jnj/testcases/BenadrylUsesTest.java]: @goutamnaik007 priority value repeated.

**Review by vaibhavNagvekarSJI — CHANGES_REQUESTED** [review]
done with the code review. Pls fix the review comments

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java (hunk 1 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/SignUpPage.java:166]: yes. After merge there were two copies of the same. So, I removed one of it from the middle
@vaibhavNagvekarSJI

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java (hunk 2 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java]: I have removed it @vaibhavNagvekarSJI

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java (hunk 3 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java]: Removed @vaibhavNagvekarSJI

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java (hunk 4 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java:256]: Removed @vaibhavNagvekarSJI

File: BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java (hunk 5 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/main/java/com/jnj/pageobjects/BenadrylUsesPage.java:279]: Removed @vaibhavNagvekarSJI

File: BenadrylAutomation/src/test/java/com/jnj/testcases/BenadrylUsesTest.java (hunk 6 above)
Comment (goutamnaik007) [reply BenadrylAutomation/src/test/java/com/jnj/testcases/BenadrylUsesTest.java]: Priorities of the test and the tests following it are updated now.
@vaibhavNagvekarSJI

**Review by vaibhavNagvekarSJI — APPROVED** [review]
reviewed done

**Pull request merged**